packages/common/src/hooks/useListData/useListData.ts

```diff
@@ -0,0 +1,75 @@
+import { SortBy } from './../useSortBy/useSortBy';
+import { UseMutateFunction } from 'react-query';
+import { useQueryClient, useMutation, useQuery } from 'react-query';
+import { QueryParams, useQueryParams } from '../useQueryParams';
+import { SortRule } from '../useSortBy';
+
+export interface ListApi<T> {
+  onQuery: ({
+    first,
+    offset,
+    sortBy,
+  }: {
+    first: number;
+    offset: number;
+    sortBy: SortBy<T>;
+  }) => () => Promise<{ data: T[]; totalLength: number }>;
+  onDelete: (toDelete: T[]) => Promise<void>;
+  onUpdate: (toUpdate: T) => Promise<T>;
+}
```

Comment:
Not a great name and this idea needs iterating on.. the idea is to have an object for each list view which has an onQuery, onDelete and onUpdate, possibly also an onCreate